### PR TITLE
validate healthcheck params in daemon side

### LINKED ISSUE #30202: Docker Daemon does not validate healthcheck param
**Description**

I create a container via two ways, one is CLI, the other is API.

```
root@ubuntu:~# docker run -d --health-cmd "curl localhost:3000" --health-interval=-1s python-demo:v1.1
docker: --health-interval cannot be negative.
See 'docker run --help'.
```
**Client side will validate the healthcheck params.**

When I use API to create a container with an **invalid parameter of negative interval**,
```
POST 192.168.59.103:2375/containers/create

{
	"Image":"python-demo:v1.1",
	"Healthcheck":{
		"Interval": -100000000
	}
}
```

Docker Daemon works fine for this, and returns the response:
```
{
  "Id": "21d712248a1fe21e72c8198b3a0b2d784a9657725e180268c40db859e178b67b",
  "Warnings": null
}
```

I think the docker daemon should validate the healthcheck params as well to provide robust API.

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
root@ubuntu:~# docker version
Client:
 Version:      1.13.0-rc4
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   88862e7
 Built:        Fri Dec 16 22:59:15 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-rc4
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   88862e7
 Built:        Fri Dec 16 22:59:15 2016
 OS/Arch:      linux/amd64
 Experimental: false
root@ubuntu:~#
```

**Output of `docker info`:**

```
root@ubuntu:~# docker info
Containers: 8
 Running: 6
 Paused: 0
 Stopped: 2
Images: 37
Server Version: 1.13.0-rc4
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 202
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: avzbj8l22g0rfl4ei2b1tm5zq
 Is Manager: false
 Node Address: 192.168.59.103
 Manager Addresses:
  192.168.59.104:2377
  192.168.59.105:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 51371867a01c467f08af739783b8beafc154c4d7
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.19.0-25-generic
Operating System: Ubuntu 14.04.3 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.954 GiB
Name: ubuntu
ID: Q2ZC:GWDN:27OH:GRMH:G6QU:W7QP:4TIX:Q5F6:YEVK:45XP:EXHC:HOB5
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 68
 Goroutines: 200
 System Time: 2017-01-17T01:11:48.938844162+08:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 https://a.b.c/
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: allencloud <allen.sun@daocloud.io>

fixes #30202 

validate user input healthcheck parameters via API in the daemon side. 

**- What I did**
1. validate healthcheck in daemon side
2. add a test for api testing invalid healthcheck parameters
3. add missing `--health-retries` validation in client side

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/container.go [hunk 1]
```diff
@@ -237,6 +237,21 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
 				return nil, err
 			}
 		}
+
+		// Validate the healthcheck params of Config
+		if config.Healthcheck != nil {
+			if config.Healthcheck.Interval < 0 {
```
Comment (cpuguy83) [inline daemon/container.go]: I think this should validate less than `1s`, wdyt as I don't think we can guarantee sub-second checking.

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: daemon/container.go (hunk 1 above)
Comment (allencloud) [inline daemon/container.go]: If that, maybe we need to use a consistent validation in the whole procedure. I think there are three places to validate:
1. client side;
2. daemon received parameters from API;
3. parameters from dockerfile when building

I think consistency is the first thing(no matter less than 0 or less than 1s), the second thing to add more specific explanation in docs. 

But currently, I am on less than 0.  😄 Open for input.

**Comment by LK4D4** [comment]
@cpuguy83 there is question for you here :)

File: daemon/container.go (hunk 1 above)
Comment (cpuguy83) [inline daemon/container.go]: Sure, consistent is good, but the daemon side is the "important" piece.
I just think if we are going to add a validation, let's make sure it's not letting non-sensical values.

**Comment by allencloud** [comment]
Hi @cpuguy83 @LK4D4 
Thanks for the review. Currently I updated this PR with modifying the validation from `0` to `1s`. And in additional, I added more specific details in docs swagger.yml and update api 1.24, since healthcheck is introduced in docker 1.12.x and api 1.24. PTAL, thanks 😄

**Comment by LK4D4** [comment]
LGTM

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**